packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx

```diff
@@ -78,8 +78,13 @@ function GridFilterInputMultipleSingleSelect(props: GridFilterInputMultipleSingl
     if (!Array.isArray(item.value)) {
       return [];
     }
-    return item.value;
-  }, [item.value]);
+
+    return item.value
+      ?.map((value) =>
+        resolvedValueOptions?.find((resolvedValue) => getOptionValue(resolvedValue) === value),
+      )
+      .filter((value) => value !== undefined) as ValueOptions[];
+  }, [getOptionValue, item.value, resolvedValueOptions]);
```

Comment:
Maybe do in one go?
```diff
diff --git a/packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx b/packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx
index 517860549..1d274b335 100644
--- a/packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx
+++ b/packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx
@@ -79,11 +79,15 @@ function GridFilterInputMultipleSingleSelect(props: GridFilterInputMultipleSingl
       return [];
     }
 
-    return item.value
-      ?.map((value) =>
-        resolvedValueOptions?.find((resolvedValue) => getOptionValue(resolvedValue) === value),
-      )
-      .filter((value) => value !== undefined) as ValueOptions[];
+    return item.value.reduce<ValueOptions[]>((acc, value) => {
+      const resolvedValue = resolvedValueOptions.find(
+        (v) => getOptionValue(v) === value,
+      );
+      if (resolvedValue != null) {
+        acc.push(resolvedValue);
+      }
+      return acc;
+    }, [] as ValueOptions[]);
   }, [getOptionValue, item.value, resolvedValueOptions]);
 
   const handleChange = React.useCallback<
```

Reply:
The adjustment has been made @MBilalShafi, thanks for the review.